federation-js/src/composition/validate/postComposition/__tests__/keyFieldsMissingOnBase.test.ts

```diff
@@ -76,6 +75,12 @@ describe('keyFieldsMissingOnBase', () => {
       Array [
         Object {
           "code": "KEY_FIELDS_MISSING_ON_BASE",
+          "locations": Array [
+            Object {
+              "column": 13,
```

Comment:
I'm trying to validate these changes, but I'm confused what this location is supposed to refer to. I may be missing something about this PR in general. Are these locations in the composed schema SDL or one of the service SDLs? Looking at [the code that generates the error](https://github.com/apollographql/federation/blob/fbbc85e1adc974c7937929534c691f74923ea87c/federation-js/src/composition/validate/postComposition/keyFieldsMissingOnBase.ts#L41), the location seems to be relative to the document resulting from parsing `@key(fields:)`, and I'm not sure how that is mapped to either of those?

Reply:
@martijnwalraven In this case, column 13, line 1 would seem to map to the left-most part of `@key` in `serviceA` above, right? e.g., pointing to the site of the directive on the base `Product` type where the omitted `uid` is?

I think the use-case here is to show users where their composition errors (as defined by the message) exist **on a subgraph** in the Studio UI?  (Note that may require #705?)

Reply:
Running the tests changes the column to 6 though, so I'm not sure how that relates to this. Still wondering how the location in the parsed field set would be related to the location in the overall document. Leaving that aside, it seems to me error locations are not useful without mapping them to individual service SDLs, so shouldn't we resolve that first before landing this?

Reply:
> Running the tests changes the column to 6 though
Is this test out of date?

> Leaving that aside, it seems to me error locations are not useful without mapping them to individual service SDLs, so shouldn't we resolve that first before landing this?
I opened an issue to add subgraph name as structured data to composition errors on the `extensions` field of `GraphQLError`. This PR exposes locations, and 90% of the time the locations array has a single location which maps to a single service that can be parsed from the error message. For `ENUM_MISMATCH` there are 2 service names in the error message, and in that case the locations correspond to one service each. 


Is it acceptable to assume the location in locations maps to the service name in the error message for this PR?

Would the best move forward be to 
1. merge this PR and open a separate PR to add subgraph name as structured data.
2. edit this PR to include subgraph names as structured data

Reply:
> This PR exposes locations, and 90% of the time the locations array has a single location which maps to a single service that can be parsed from the error message. For ENUM_MISMATCH there are 2 service names in the error message, and in that case the locations correspond to one service each.

If we're already relying on parsing error messages, merging this and solving structured error metadata separately seems fine. I do think depending on the specific contents of error messages is dangerous, and limits our ability to evolve composition validation, so we should move away from that. But we can talk about that as part of #705.